modules/standard/List.chpl

```diff
@@ -764,6 +764,26 @@ module List {
 
       return result;
     }
+    
+    proc const ref first() const ref {
```

Comment:
@mppf Do you think it would be OK for all these overloads to call out to a single `proc ref _first() ref;` implementation?

Reply:
They could call a single implementation but it can't have `ref` `this` intent because that wouldn't compile. It'd have to be a helper method that used `const ref` `this` intent and returned a `ref`.

Reply:
@vijaya-bhargavi Would you be up for trying to use a helper method here to avoid code duplication? You'd want to call it in both the `ref` and `const ref` receiver overloads.